spring-webmvc/src/test/java/org/springframework/web/servlet/i18n/LocaleResolverTests.java

```diff
@@ -99,6 +99,8 @@ private void doTest(LocaleResolver localeResolver, boolean shouldSet) {
 
 			if (localeContextResolver instanceof AbstractLocaleContextResolver) {
 				((AbstractLocaleContextResolver) localeContextResolver).setDefaultTimeZone(TimeZone.getTimeZone("GMT+1"));
+				request.removeAttribute(CookieLocaleResolver.LOCALE_REQUEST_ATTRIBUTE_NAME);
+				localeContextResolver.resolveLocaleContext(request);
```

Comment:
Extending `AbstractLocaleContextResolver` uncovered some test scenarios that weren't asserted for `CookieLocaleResolver`. This workaround isn't pretty, but is needed since `CookieLocaleResolver` consumes default locale while resolving locale context.